src/network/networkctl.c

```diff
@@ -2919,6 +2930,388 @@ static int verb_reconfigure(int argc, char *argv[], void *userdata) {
+                                return r;
+
+                        r = add_config_to_edit(&context, path, dropins);
+                        if (r < 0)
+                                return r;
+
+                        /* We only operate on .network file here, so no need to check or trigger udevd */
+                        networkd_reload = true;
+
+                        continue;
+                }
+
+                if (ENDSWITH_SET(*name, ".network", ".netdev"))
+                        networkd_reload = true;
+                else if (endswith(*name, ".link"))
+                        udevd_trigger = true;
+                else
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid network config name '%s'.", *name);
+
+                r = get_config_files_by_name(*name, &path, &dropins);
+                if (r == -ENOENT) {
+                        if (arg_drop_in)
+                                return log_error_errno(r, "Cannot find network config '%s'.", *name);
+
+                        log_debug("No existing network config '%s' found, creating a new file.", *name);
+
+                        path = path_join(NETWORK_DIRS[0], *name);
+                        if (!path)
+                                return log_oom();
+
+                        r = edit_files_add(&context, path, NULL, NULL);
+                        if (r < 0)
+                                return r;
+                        continue;
+                }
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get the path of network config '%s': %m", *name);
+
+                r = add_config_to_edit(&context, path, dropins);
+                if (r < 0)
+                        return r;
+        }
+
+        r = do_edit_files_and_install(&context);
+        if (r < 0)
+                return r;
+
+        if (arg_no_reload)
+                return 0;
+
+        if (!sd_booted() || running_in_chroot() > 0) {
+                log_debug("System is not booted with systemd or is running in chroot, skipping reload.");
+                return 0;
+        }
+
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+
+        r = sd_bus_open_system(&bus);
+        if (r < 0)
+                return log_error_errno(r, "Failed to connect to system bus: %m");
+
+        if (udevd_trigger) {
+                r = udevd_reload(bus);
+                if (r < 0)
+                        return r;
+                if (r > 0) {
+                        r = safe_fork("(udevadm)",
+                                      FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG|FORK_LOG|FORK_WAIT|FORK_FLUSH_STDIO,
+                                      NULL);
+                        if (r < 0)
+                                return r;
+                        if (r == 0) { /* Child */
+                                execlp("udevadm", "udevadm", "trigger", "--subsystem-match=net", "--action=add",
```

Comment:
also, this should be action "change", not "add". "add" really should be about new devices only, not about devices that are changed.

Reply:
> i'd prefer if we implement this locally, i.e. enumerate network devices via sd_device_enumerator_add_match_subsystem(), then issue sd_device_trigger_with_uuid() and wait for the events to be delivered back to us, identified by the id, as reported by sd_device_get_trigger_uuid()

I think that would be an overkill, i.e. we would just reimplement (copy over) udevadm-trigger.c here 🤔

Reply:
> also, this should be action "change", not "add". "add" really should be about new devices only, not about devices that are changed.

Hmm, the same documentation still: https://github.com/systemd/systemd/blob/b9dac418372401742609bd600f05267ae3a724de/man/systemd.link.xml#L1151

Reply:
Hmm... Please do not trigger all network interfaces, but only trigger interfaces that matches or previously matched with the .link file.

Reply:
> Hmm... Please do not trigger all network interfaces, but only trigger interfaces that matches or previously matched with the .link file.

But we are not aware of that if we're given config names?

(Also, would like to know which action to use)

Reply:
> (Also, would like to know which action to use)

'add' should be used, otherwise, most settings will not be applied.

Reply:
> But we are not aware of that if we're given config names?

networkd does more gracefully.
Maybe,
1. export the .link file parser,
2. enumerate network interfaces,
3. check which .link file will be applied to each interface,
4. if the matching file for an interface is different from the previous, or is one of the files we have edited, then trigger an 'add' uevent for the interface.

Reply:
This sounds like something that deserves its own PR 😉

I mean, if we're doing this, it might be even better to issue `reconfigure` for the changed `.network` configs. But that's too much complexity and makes `networkctl` somewhat responsible for config parsing... Not sure if that's really worth it.